controllers/clustergroupupgrade_controller.go

```diff
@@ -765,14 +784,25 @@ func (r *ClusterGroupUpgradeReconciler) doManagedPoliciesExist(
 			continue
 		}
 
+		for _, policyT := range childPolicy.Spec.PolicyTemplates {
+			configPolicy := &unstructured.Unstructured{}
+			json.Unmarshal(policyT.ObjectDefinition.Raw, configPolicy)
+
+			configPlcTAnnotations := configPolicy.GetAnnotations()
+			// Identify policies have invalid hub templates
+			_, ok := configPlcTAnnotations["policy.open-cluster-management.io/hub-templates-error"]
+			if ok || strings.Contains(string(policyT.ObjectDefinition.Raw), "{{hub") {
```

Comment:
Is the string '{{hub...' part of the value from the policy.open-cluster-management.io/hub-templates-error annotation? If so it may make sense to restrict the string search to just the value rather than the whole config policy.

Reply:
Good catch! Shouldn't check on the whole config policy as the annotation has "{{hub". Will update

Reply:
Ah, I missed that you are checking the child policy. Is the intent here that if the _child_ policy contains "{{hub" then ACM failed to do the template substitution and the child policy is thus invalid?
If so, is this a test that is periodically re-evaluated?

Reply:
Yes, if the child policy contains "{{hub", it means the hub template processing failed on the hub and the root policy is invalid. The hub templating is processed on the hub when policies are created or updated during propogation of repliciated policies to spoke clusters. The hub-template-error annotation should be added to the child policy by the propogator on the hub if there has invalid template. In case the annotation was not added,  the config policy controller running on the spoke cluster also checks if child policy has '{{hub' and adds the annotation, so we are unlikely to have the situation where the policy contains invalid template without the error annotation attached. Just in case something weird happens, I check the presence of '{{hub' against child policy here as well. 

It's re-evaluated if CGU is not ready to be remediated due to any reason, ie. not all resources have been prepared, Cgu is not enabled etc

Reply:
I wonder if it would be more efficient to simply look for the string every time? In the current implementation we are unmarshalling the entire policy every time to do this check and then, if there is not an error (no annotation) we scan the entire policy for a string comparison.

Reply:
Good point. Yep, we actually just need check the string {{hub. Will update.